src/main/java/kafka/vertx/demo/WebSocketServer.java

```diff
@@ -41,22 +42,8 @@ public void start(Promise<Void> startPromise) {
     router.get().handler(StaticHandler.create());
 
     loadKafkaConfig()
-      .onSuccess(config -> {
-        kafkaConfig = new HashMap<>();
-        config.forEach(entry -> kafkaConfig.put(entry.getKey(), entry.getValue().toString()));
-        vertx.createHttpServer()
-          .requestHandler(router)
-          .webSocketHandler(this::handleWebSocket)
-          .listen(8080, "localhost")
-          .onSuccess(ok -> {
-            logger.info("🚀 WebSocketServer started");
-            startPromise.complete();
-          })
-          .onFailure(err -> {
-            logger.error("❌ WebSocketServer failed to listen", err);
-            startPromise.fail(err);
-          });
-      })
+      .compose(config -> startWebSocket(router, config))
+      .onSuccess(ok -> startPromise.complete())
```

Comment:
if you're not logging the ok, might as well just use startPromise::complete

Reply:
I'm guessing this works - which means the new vertx is allowing multiple onsuccess (you add one to the future in startwebsocket, and add one here too)

Reply:
Yeah it does seem to call both, also works for multiple onFailure.

For the startPromise completion if I use `startPromise::complete` it won't compile, it fails with `incompatible types: io.vertx.core.http.HttpServer cannot be converted to java.lang.Void`

Reply:
Fair point, forgot it would complete with that value. You could use `.mapEmpty().onSuccess(startPromise::complete)` - your call if that makes it seem more intentional?

Reply:
I did try that as well but since HttpServer gets mapped to an empty Object and not Void it still complains